lib/Service/BanService.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Talk\Service;
+
+use DateTime;
+use OCA\Talk\Model\Ban;
+use OCA\Talk\Model\BanMapper;
+
+class BanService {
+	
+	public function __construct(
+		protected BanMapper $banMapper,
+	) {
+	}
+
+	/**
+	 * Validate the ban data.
+	 */
+	private function validateBanData(string $actorId, string $actorType, int $roomId, string $bannedByActorId, string $bannedByActorType, ?DateTime $bannedAt, ?string $reason): void {
+		if (empty($actorId) || empty($actorType) || empty($roomId) || empty($bannedByActorId) || empty($bannedByActorType)) {
+			throw new \InvalidArgumentException("Invalid ban data provided.");
+		}
+
+		if ($bannedAt !== null && !$bannedAt instanceof DateTime) {
+			throw new \InvalidArgumentException("Invalid date format for bannedAt.");
+		}
+	}
+
+	/**
+	 * Create a new ban
+	 */
+	public function createBan(string $actorId, string $actorType, int $roomId, string $bannedByActorId, string $bannedByActorType, ?DateTime $bannedAt, ?string $reason): Ban {
+		$this->validateBanData($actorId, $actorType, $roomId, $bannedByActorId, $bannedByActorType, $bannedAt, $reason);
+
+		$ban = new Ban();
+		$ban->setActorId($actorId);
+		$ban->setActorType($actorType);
+		$ban->setRoomId($roomId);
+		$ban->setBannedByActorId($bannedByActorId);
+		$ban->setBannedByActorType($bannedByActorType);
+		$ban->setBannedAt($bannedAt ?? new DateTime());
```

Comment:
Because it is not easily testable, we have `OCP\AppFramework\Utility\ITimeFactory::getDateTime()` to generate "now"